what4/src/What4/Protocol/VerilogWriter/AST.hs

```diff
@@ -265,8 +271,8 @@ data Item where
 -- | Necessary monadic operations
 
 data ModuleState sym n =
-    ModuleState { vsInputs :: [(Some WT.BaseTypeRepr, Identifier)]
-                -- ^ All module inputs, in reverse order.
+    ModuleState { vsInputs :: [(Word64, Some WT.BaseTypeRepr, Identifier)]
+                -- ^ All module inputs, in reverse order. We use Word64 for Nonces to avoid GHC bugs.
```

Comment:
A link or more information on the GHC bugs might be good here, especially if we expect those bugs to be fixed at some point.

Reply:
Good point. I'll add links.